CIP-0XXX/README.md

```diff
@@ -0,0 +1,551 @@
+part `cardano-node`), then we wouldn't have that opportunity because `Cardano`
+must meet its deadlines, or the security of the system as a whole must be at
+stake.
+
+Eventually, there will be many `Mithril` applications, and all of them might
+have different network requirements and thus might require slightly different
+configurations.  We SHOULD aim to build something that can be used as
+a scaffolding for such applications. This may also open future avenues for
+delivering new functionalities that fit together well with existing
+infrastructure while still mitigating the risk of catastrophic events at the
+core of the system.
+
+Please also note that any protocols and their instances that will be built as
+part of the standalone `Mithril` node could be reused in future for `Peras` and
+`Leios`.  It will give us even more confidence that the future core system will
+be built from components that are used in production.
+
+`Cardano` and `Mithril`, as separate executables, can still be packaged
+together, lowering the barrier of participation. When run separately, the SPO
+is also in better control of the resources dedicated to each process - this is
+important for the health of both systems.
+
+Another benefit of such a design is that `Mithril` can be developed on its own
+pace, not affected by significant changes in other parts of the system like
+ledger - the Cardano Core team restrains itself to not publishing new `cardano-node`
+versions with significant changes across many of its subsystems - just for the
+sake of clarity of where to look if a bug is found.
+
+In this design, a `Mithril` node runs alongside a `cardano-node`, which is connected to
+it via a UNIX socket (`node-to-client`) protocol.  This means an SPO will run
+a `Mithril` node per `cardano-node`.  The `Mithril` node connected to BP can
+also have access to necessary keys for signing purposes. `Mithril` might also
+use the KES agent, as `cardano-node` will in the near future, to securely access
+the KES key.
+
+
+# Path to Active
+
+## Acceptance Criteria
+
+1. A Cardano node implementing the previously described mini-protocols is released for production.
+1. A message producer node (e.g. a Mithril signer) publishing messages to the local Cardano node through mini-protocols is released.
+1. A message subscriber node (e.g. a Mithril aggregator) receiving messages from the local Cardano node is released.
+
+## Implementation Plan
+
+> [!WARNING]  
+> A hard-fork of the Cardano chain may be required.
+
+* Write a "formal" specification of the protocols along with vectors/conformance checker for protocol's structure and state machine logic.
+* Validate protocol behaviour with all relevant parties (Network and Node teams).
+* Validate performance profile and impact on Cardano network.
+* Implement the Cardano n2n and n2c mini-protocols in the Cardano node.
+* Implement the Cardano n2c mini-protocols in Mithril signer and aggregator nodes.
```

Comment:
I would add, write a contained document describing the designed architecture, essentially extending this CIP with more technical or details about the solution

Reply:
I added a bullet point for a written architecture document